dsm/resource_plugin.go

```diff
@@ -0,0 +1,234 @@
+// **********
+// Terraform Provider - SDKMS: resource: security object
+// **********
+//       - Author:    Ravi Gopal at fortanix dot com
+//       - Version:   0.5.1
+//       - Date:      21/10/2022
+// **********
+
+package dsm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+ )
+
+// [-] Define Plugin
+func resourcePlugin() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCreatePlugin,
+		ReadContext:   resourceReadPlugin,
+		UpdateContext: resourceUpdatePlugin,
+		DeleteContext: resourceDeletePlugin,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"plugin_id": {
+				Type:     schema.TypeString,
+				 Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default: "",
+			},
+			"plugin_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default: "standard",
+			},
+			"default_group": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"groups": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"language": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default: "LUA",
+			},
+			"code": {
+				 Type:     schema.TypeString,
+				 Required: true,
+			 },
+			 "enabled": {
+				 Type:     schema.TypeBool,
+				 Optional: true,
+				 Default: true,
+			 },
+			 "acct_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creator": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+var endpoint = "sys/v1/plugins"
```

Comment:
You haven't implemented the situation when the group(s) set for the plugin have an approval policy. In such a case the endpoint changes to "sys/v1/approval_requests" and the payload format changes to:
{
	"method": "POST",
	"operation": "/sys/v1/plugins",
	"body": {
		"name": "ccc1",
		"description": "ccc_desc",
		"plugin_type": "standard",
		"default_group": "67d715ca-cf0e-4daf-b6c6-014a52254cb6",
		"source": {
			"language": "lua",
			"code": "function run(input)\n  test\nend"
		},
		"add_groups": ["67d715ca-cf0e-4daf-b6c6-014a52254cb6"]
	}
}

Reply:
@darktower When a customer uses username/password, does approval_request really matters?
Because using postman and terraform I can able to create a plugin even after groups having the approval policy.

Reply:
Added the approval_requests.